src/modules/cmdpal/Microsoft.CmdPal.Core.ViewModels/FiltersViewModel.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using CommunityToolkit.Mvvm.ComponentModel;
+using Microsoft.CmdPal.Core.ViewModels.Models;
+using Microsoft.CommandPalette.Extensions;
+
+namespace Microsoft.CmdPal.Core.ViewModels;
+
+public partial class FiltersViewModel : ExtensionObjectViewModel
+{
+    private readonly ExtensionObject<IFilters> _filtersModel = new(null);
+
+    [ObservableProperty]
+    public partial string CurrentFilterId { get; set; } = string.Empty;
+
+    [ObservableProperty]
+    [NotifyPropertyChangedFor(nameof(ShouldShowFilters))]
+    public partial IFilterItemViewModel[] Filters { get; set; } = [];
+
+    public bool ShouldShowFilters => Filters.Length > 0;
+
+    public FiltersViewModel(ExtensionObject<IFilters> filters, WeakReference<IPageContext> context)
+        : base(context)
+    {
+        _filtersModel = filters;
+    }
+
+    public override void InitializeProperties()
+    {
+        if (_filtersModel is not null)
```

Comment:
I don't think `_filtersModel` will ever be null, but `_filtersModel.Unsafe` might be?